addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js

```diff
@@ -1200,6 +1200,13 @@ X[]
                         contentAfter: '<p>ab&nbsp;[]</p>',
                     });
                 });
+                it('should not merge the paragraph into h1', async () => {
+                    await testEditor(BasicEditor, {
+                    contentBefore: '<h1>[abc</h1><h1>def]<br></h1><p>ghi</p>',
```

Comment:
Did this bug only happen when the second `h1` ended with a `br` ?

Reply:
NO, its not specifically for the second or third h1 but our case is when we hit backspace after selecting text the next line which has its style or tag should not convert to the upper tag upto where selection is performed.

Reply:
So you would have had the bug with `<h1>[abc</h1><p>d]ef</p>`, where you'd end up with `<h1>[]ef</h1>` instead of `<p>[]ef</p>` as well ? If so maybe test that instead.

Reply:
Shouldn't this test case be more like this ?
`<h1>[abc</h1><p>g]hi</p>` => backspace =>  `<p>hi</p>`
And it should fail without your fix because it would result in `<h1>h1</h1>` which is wrong.

Reply:
@dmo-odoo this test case i worked on was as per your specifications .Here are we thinking of replacing the tag of text at focus to the tag of text at anchor during selection ?
![Screenshot from 2022-12-09 17-51-36](https://user-images.githubusercontent.com/103405229/206703025-9eeee542-7b76-4dbd-b9f2-710abcd213b5.png)

Reply:
@shsa-odoo Well not really beause there was not a `<br>` in my test case. Also, I realized while discussing with @Zinston that the test case I gave you was incomplete, therefore I would like you to replace it with the tests I mentionned above, thanks !